library/pk_wrap.c

```diff
@@ -636,15 +636,9 @@ static int eckey_verify_wrap(void *ctx, mbedtls_md_type_t md_alg,
                              const unsigned char *sig, size_t sig_len)
 {
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
-    mbedtls_ecdsa_context ecdsa;
-
-    mbedtls_ecdsa_init(&ecdsa);
+    mbedtls_ecp_keypair *keypair_ctx = ctx;
```

Comment:
This seems redundant now, we could just pass `ctx` directly.

Or actually, thinking while writing, do we really need two sets of functions (one for `eckey`, one for `ecdsa`)?

Reply:
well, this is something I also thought while working on this. Perhaps the answer might be no, so I'll check that